tcan_can_j1939/include/tcan_can_j1939/J1939CanMsg.hpp

```diff
@@ -36,6 +38,25 @@ class J1939CanMsg : public tcan_can::CanMsg {
 
     constexpr uint32_t getParameterGroupCanId() const { return getCobId() & CAN_ID_PGN_MASK; }
 
+    friend std::ostream& operator<<(std::ostream& os, const J1939CanMsg& msg) {
+        os << "Priority: " << +msg.getPriority() << std::endl
+           << "ExtendedDataPage: " << +msg.getExtendedDataPage() << std::endl
+           << "DataPage: " << +msg.getDataPage() << std::endl
+           << "PduFormat: " << +msg.getPduFormat() << std::endl
+           << "PduSpecific: " << +msg.getPduSpecific() << std::endl
+           << "SourceAddress: " << +msg.getSourceAddress() << std::endl
+           << "ParameterGroupNumber: " << +msg.getParameterGroupNumber() << std::endl
+           << "ParameterGroupCanId: " << +msg.getParameterGroupCanId() << std::endl
+           << "Data: ";
+
+        for (int idx = 0; idx < msg.getLength() ; ++idx) {
```

Comment:
very very minor change about style

```suggestion
        for (int idx = 0; idx < msg.getLength(); ++idx) {
```